source/MaterialXTest/ShaderValid.cpp

```diff
@@ -498,67 +503,100 @@ TEST_CASE("GLSL geometry", "[shadervalid]")
     nodeGraph->removeNode(image1->getName());
 }
 
-bool endsWithCaseInsensitive(std::string str, std::string toMatch)
+void getSubDirectories(std::string& baseDirectory, mx::StringVec& relativePaths)
 {
-    auto it = toMatch.begin();
-	  return str.size() >= toMatch.size() && std::all_of(std::next(str.begin(),str.size() - toMatch.size()), str.end(),
-        [&it](const char & c)
+    relativePaths.push_back(baseDirectory);
```

Comment:
I don't think this will occur we can run into the same directory more than once, if there are symbolic links but let's assume it does not occur.